ADAL/src/urlprotocol/ADURLProtocol.m

```diff
@@ -139,7 +140,10 @@ + (NSURLRequest *)canonicalRequestForRequest:(NSURLRequest *)request
 {
     AD_LOG_VERBOSE_F(@"+[ADURLProtocol canonicalRequestForRequest:]", _reqCorId(request), @"host: %@", [request.URL host] );
     
-    return request;
+    NSMutableURLRequest* mutableRequest = [request mutableCopy];
```

Comment:
This change makes me nervous. Like, very nervous. Before this method effectively did nothing but log out a bit. Now it's creating whole new request objects. The interaction between WebKit and NSURLProtocol is not particularly well understood and varies from OS version to OS version. Also the purpose of canonicalRequestForRequest is to do URL canonicalization, which is not happening here so that adds to the nervousness that it won't have the desired effect, or worse break with an OS update.

Is it at all possible to fix this bug without changing ADURLProtocol at all? Can we do this in -startLoading instead?